### DISCUSSION TITLE
how to retry request by custom policy

### DISCUSSION THREAD

**Discussion by xj524598** [hyperium/hyper#3489] [Q&A]
pseudo code：
loop {
let resp=client.execute(request);
//custom policy
if current_time="2023-11-11"{
continue;
}
break;
}
if want retry like this,request must clone, request has no clone method

**Comment by xj524598** [comment]
even request has clone() method, i think it may cost expansive.
if we can use send_request method,we can get request from ClientError

**Comment by OldMansKid** [comment]
In my experience, this isn't a problem. Even if you are posting a lot of data, your data must live somewhere and if the request fails, you can simply reconstruct the headers and stream the data to the server again, unless you're writing a proxy and is streaming the data from the client to the origin server. In that case, I don't think it's possible to retry sending the request.
